packages/oas-utils/src/transforms/import-spec.ts

```diff
@@ -134,12 +132,12 @@ export async function importSpecToWorkspace(
       tags: Tag[]
       securitySchemes: SecurityScheme[]
     }
-  | { error: true; importWarnings: string[] }
+  | { error: true; importWarnings: string[]; collection: undefined }
 > {
   const { schema, errors } = await parseSchema(spec, { shouldLoad })
   const importWarnings: string[] = [...errors.map((e) => e.message)]
 
-  if (!schema) return { importWarnings, error: true }
+  if (!schema) return { importWarnings, error: true, collection: undefined }
```

Comment:
there might be a better solution, but without it TS is complaining when I write tests like this:

```
expect(result.collection?.info).toEqual({
  title: 'API',
})
```

Reply:
yes in your test you need to pass the type guard first! 

https://github.com/scalar/scalar/blob/main/packages/api-client/src/libs/send-request/create-request-operation.test.ts#L132

we should actually switch to using this error response style here as well